src/rpc/methods/f3/types.rs

```diff
@@ -293,6 +297,101 @@ pub struct F3Manifest {
 }
 lotus_json_with_self!(F3Manifest);
 
+impl F3Manifest {
+    pub fn get_eth_return_from_message_receipt(receipt: &Receipt) -> anyhow::Result<Vec<u8>> {
+        anyhow::ensure!(
+            receipt.exit_code().is_success(),
+            "unsuccessful exit code {}",
+            receipt.exit_code()
+        );
+        let return_data = receipt.return_data();
+        let eth_return =
+            fvm_ipld_encoding::from_slice::<fvm_ipld_encoding::BytesDe>(&return_data)?.0;
+        Ok(eth_return)
+    }
+
+    #[allow(clippy::indexing_slicing)]
```

Comment:
Wouldn't it be safer to use `get`? Especially if the data structure changes, we don't want Forest to crash.

Reply:
Fixed.